### PR TITLE
Update custom .NET Core host tutorial for hosting with hostfxr

### PR DESCRIPTION
**Pull request by elinor-fung**
## Summary

Update custom .NET Core host tutorial for hosting with hostfxr

See dotnet/samples#995 for snippet tags.

### CODE REVIEW

**Comment by rpetrusha** [comment]
Closing and reopening to begin new build.

**Comment by elinor-fung** [comment]
Hm, I must be missing / misunderstanding something with how to use the snippet tags and code blocks. The ones I tried to add are blank in https://review.docs.microsoft.com/en-us/dotnet/core/tutorials/netcore-hosting?branch=pr-en-us-12872

**Comment by elinor-fung** [comment]
Actually, from the logs, it looks like the build was still using the samples repo at commit 8eda260763f8f7441bfcdff89f3daf1f810f5f33 which is from before the change for adding the snippet tags got merged?

**Comment by rpetrusha** [comment]
I can't see anything wrong, @elinor-fung. The previous build (before I closed and reopened this PR) reported no warnings, although it should have. But I'll close and reopen once again, and we can see if that accomplishes anything.

**Comment by rpetrusha** [comment]
The code is still not included in the build, but there are no build warnings. @mairaw, I may well be missing something completely obvious, but could you take a look, please? The samples repo contains the code, and the code references and snippet tags are correct, AFAIK.

**Comment by rpetrusha** [comment]
Closing and reopening again...

**Comment by elinor-fung** [comment]
Thanks, @rpetrusha. It seems like it picked up the changes from the samples repo this time.

**Comment by rpetrusha** [comment]
Yes, it finally did, @elinor-fung. I have no idea of what the problem was earlier. I'll review your PR shortly.

File: docs/core/tutorials/netcore-hosting.md [hunk 1]
```diff
@@ -20,20 +20,62 @@ Because hosts are native applications, this tutorial will cover constructing a C
...
+
+The following steps detail how to use the `nethost` and `hostfxr` libraries to start the .NET Core runtime in a native application and call into a managed static method. The [sample](https://github.com/dotnet/samples/tree/master/core/hosting/HostWithHostFxr) uses the `nethost` header and library installed with the .NET SDK and copies of the [`coreclr_delegates.h`](https://github.com/dotnet/core-setup/blob/master/src/corehost/cli/coreclr_delegates.h) and [`hostfxr.h`](https://github.com/dotnet/core-setup/blob/master/src/corehost/cli/hostfxr.h) files from the [dotnet/core-setup](https://github.com/dotnet/core-setup) repository.
+
+### Step 1 - Load HostFxr and get exported hosting functions
+
+The `nethost` library provides the `get_hostfxr_path` function for locating the `hostfxr` library. The `hostfxr` library exposes functions for hosting the .NET Core runtime. The full list can be found in [`hostfxr.h`](https://github.com/dotnet/core-setup/blob/master/src/corehost/cli/hostfxr.h) and the [native hosting design document](https://github.com/dotnet/core-setup/blob/master/Documentation/design-docs/native-hosting.md). The sample and this tutorial use the following:
```
Comment (rpetrusha) [inline docs/core/tutorials/netcore-hosting.md]:
```suggestion
The `nethost` library provides the `get_hostfxr_path` function for locating the `hostfxr` library. The `hostfxr` library exposes functions for hosting the .NET Core runtime. The full list of functions can be found in [`hostfxr.h`](https://github.com/dotnet/core-setup/blob/master/src/corehost/cli/hostfxr.h) and the [native hosting design document](https://github.com/dotnet/core-setup/blob/master/Documentation/design-docs/native-hosting.md). The sample and this tutorial use the following:
```

**Review by rpetrusha — CHANGES_REQUESTED** [review]
This looks really good, @elinor-fung. I've just suggested one very minor change and this will be ready to merge.

**Comment by elinor-fung** [comment]
cc @sdmaclea @vitek-karas @lpereira

**Review by sdmaclea — COMMENTED** [review]
LGTM.

For the sample code itself, I would consider renaming the `load_and_get` variable name to be more verbose. When I first saw it wasn't immediately clear what it was doing.

**Review by rpetrusha — APPROVED** [review]

**Comment by rpetrusha** [comment]
Thanks, @elinor-fung. I'll merge your PR now.

**Pull request merged**